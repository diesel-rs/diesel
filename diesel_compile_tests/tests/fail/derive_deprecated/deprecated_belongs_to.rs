#[macro_use]
extern crate diesel;

table! {
    bar (id) {
        id -> Integer,
    }
}

table! {
    foo (bar_id) {
        bar_id -> Integer,
    }
}

#[derive(Identifiable)]
#[diesel(table_name = bar)]
struct Bar {
    id: i32,
}

#[derive(Identifiable)]
#[diesel(table_name = bar)]
struct Baz {
    id: i32,
}

#[derive(Associations)]
#[belongs_to]
//~^ ERROR: unexpected end of input, expected parentheses
#[diesel(table_name = foo)]
struct Foo1 {
    bar_id: i32,
}

#[derive(Associations)]
#[belongs_to = Bar]
//~^ ERROR: attribute value must be a literal
//~| ERROR: expected parentheses
#[diesel(table_name = foo)]
struct Foo2 {
    bar_id: i32,
}

#[derive(Associations)]
#[belongs_to()]
//~^ ERROR: unexpected end of input, expected identifier
#[diesel(table_name = foo)]
struct Foo3 {
    bar_id: i32,
}

#[derive(Associations)]
#[belongs_to("what")]
//~^ ERROR: expected identifier
#[diesel(table_name = foo)]
struct Foo4 {
    bar_id: i32,
}

#[derive(Associations)]
#[belongs_to(parent)]
//~^ ERROR: unexpected end of input, expected `=`
#[diesel(table_name = foo)]
struct Foo5 {
    bar_id: i32,
}

#[derive(Associations)]
#[belongs_to(parent())]
//~^ ERROR: expected `=`
#[diesel(table_name = foo)]
struct Foo6 {
    bar_id: i32,
}

#[derive(Associations)]
#[belongs_to(parent = 1)]
//~^ ERROR: expected string literal
#[diesel(table_name = foo)]
struct Foo7 {
    bar_id: i32,
}

#[derive(Associations)]
#[belongs_to(parent = "1")]
//~^ ERROR: expected identifier
#[diesel(table_name = foo)]
struct Foo8 {
    bar_id: i32,
}

#[derive(Associations)]
#[belongs_to(Bar)]
#[diesel(table_name = foo)]
struct Foo9 {
    bar_id: i32,
}

#[derive(Associations)]
#[belongs_to(Bar, what)]
//~^ ERROR: expected `foreign_key`
#[diesel(table_name = foo)]
struct Foo10 {
    bar_id: i32,
}

#[derive(Associations)]
#[belongs_to(Bar, foreign_key)]
//~^ ERROR: unexpected end of input, expected `=`
#[diesel(table_name = foo)]
struct Foo11 {
    bar_id: i32,
}

#[derive(Associations)]
#[belongs_to(Bar, foreign_key = 1)]
//~^ ERROR: expected string literal
#[diesel(table_name = foo)]
struct Foo12 {
    bar_id: i32,
}

#[derive(Associations)]
#[belongs_to(Bar, foreign_key = "1")]
//~^ ERROR: expected identifier
#[diesel(table_name = foo)]
struct Foo13 {
    bar_id: i32,
}

#[derive(Associations)]
#[belongs_to(Baz, foreign_key = "bar_id")]
#[diesel(table_name = foo)]
struct Foo14 {
    bar_id: i32,
}

fn main() {}
