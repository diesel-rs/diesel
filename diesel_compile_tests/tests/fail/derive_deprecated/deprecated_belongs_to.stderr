error: unexpected end of input, expected parentheses
       help: The correct format looks like `#[diesel(belongs_to(Foo, foreign_key = foo_id))]`
  --> tests/fail/derive_deprecated/deprecated_belongs_to.rs:29:3
   |
29 | #[belongs_to]
   |   ^^^^^^^^^^

error: expected parentheses
  --> tests/fail/derive_deprecated/deprecated_belongs_to.rs:36:14
   |
36 | #[belongs_to = Bar]
   |              ^

error: unexpected end of input, expected identifier
  --> tests/fail/derive_deprecated/deprecated_belongs_to.rs:43:13
   |
43 | #[belongs_to()]
   |             ^^

error: expected identifier
  --> tests/fail/derive_deprecated/deprecated_belongs_to.rs:50:14
   |
50 | #[belongs_to("what")]
   |              ^^^^^^

error: unexpected end of input, expected `=`
       help: The correct format looks like `#[diesel(belongs_to(Foo, foreign_key = foo_id))]`
  --> tests/fail/derive_deprecated/deprecated_belongs_to.rs:57:14
   |
57 | #[belongs_to(parent)]
   |              ^^^^^^

error: expected `=`
  --> tests/fail/derive_deprecated/deprecated_belongs_to.rs:64:20
   |
64 | #[belongs_to(parent())]
   |                    ^

error: expected string literal
  --> tests/fail/derive_deprecated/deprecated_belongs_to.rs:71:23
   |
71 | #[belongs_to(parent = 1)]
   |                       ^

error: expected identifier
  --> tests/fail/derive_deprecated/deprecated_belongs_to.rs:78:23
   |
78 | #[belongs_to(parent = "1")]
   |                       ^^^

error: expected `foreign_key`
  --> tests/fail/derive_deprecated/deprecated_belongs_to.rs:92:19
   |
92 | #[belongs_to(Bar, what)]
   |                   ^^^^

error: unexpected end of input, expected `=`
       help: The correct format looks like `#[diesel(belongs_to(Foo, foreign_key = foo_id))]`
  --> tests/fail/derive_deprecated/deprecated_belongs_to.rs:99:19
   |
99 | #[belongs_to(Bar, foreign_key)]
   |                   ^^^^^^^^^^^

error: expected string literal
   --> tests/fail/derive_deprecated/deprecated_belongs_to.rs:106:33
    |
106 | #[belongs_to(Bar, foreign_key = 1)]
    |                                 ^

error: expected identifier
   --> tests/fail/derive_deprecated/deprecated_belongs_to.rs:113:33
    |
113 | #[belongs_to(Bar, foreign_key = "1")]
    |                                 ^^^

error: unexpected expression: `Bar`
  --> tests/fail/derive_deprecated/deprecated_belongs_to.rs:36:16
   |
36 | #[belongs_to = Bar]
   |                ^^^
